Core/include/Acts/Utilities/HelpersMatrixGenerated.ipp

```diff
@@ -0,0 +1,825 @@
+//////////////////////////////////////////////////
+// This function is AUTOGENERATED. Do not edit! //
+//////////////////////////////////////////////////
+inline ActsMatrix<8, 8> multiply(const ActsMatrix<8, 1>& A,
+                                 const ActsMatrix<1, 8>& B) {
+  ActsMatrix<8, 8> C;
+  double* pC = C.data();
+  const double* pA = A.data();
+  const double* pB = B.data();
```

Comment:
For highest-quality vectorized code when using wider hardware vectorization (which you can try out by building affected compilation units in `-march=native` mode), you will likely need to tell the compiler that these matrices have been properly aligned on the stack by Eigen. You can do this with the `__builtin_assume_aligned` intrinsic, available in both clang and GCC, which I tend to wrap in the following way for convenience:

```c++
  template <typename T>
  const T* assume_aligned(const T* ptr, size_t alignment) {
    return static_cast<const T*>(__builtin_assume_aligned(ptr, alignment));
  }
  
  template <typename T>
  T* assume_aligned(T* ptr, size_t alignment) {
    return static_cast<T*>(__builtin_assume_aligned(ptr, alignment));
  }
  ```

Eigen should impose an alignment that matches the largest hardware vector size, so 64 on AVX-512 CPUs and 32 on AVX-256 CPUs (which is the vast majority of modern hardware).